### PR TITLE
Database

### PR DESCRIPTION
**Pull request by GAlexandrD**

### CODE REVIEW

File: src/datastore/db.ts [hunk 1]
```diff
@@ -0,0 +1,128 @@
...
+import path from 'path';
+
+interface Index {
+  [key: string]: string[];
+}
+
+const ERROR_NOT_OPENED = 'Db wasn`t opened';
+const ERROR_NO_SUCH_MODEL = 'no such model';
+const ERROR_NO_SUCH_RECORD = 'no such record';
+
```
Comment (kziya) [inline src/datastore/db.ts]: import constants from another file `db-errors.enum.ts` and switch to Enum 
```ts
export enum DbErrorMessages{
    ERROR_NOT_OPENED = 'Db wasn`t opened',
    ERROR_NO_SUCH_MODEL = 'no such model',
    ERROR_NO_SUCH_RECORD = 'no such record',
}
```

File: src/datastore/taskService.ts [hunk 2]
```diff
@@ -0,0 +1,77 @@
...
+  description: string;
+  deadline: string;
+  done: string;
+}
+
+export interface ITaskP {
+  name?: string;
+  description?: string;
+  deadline?: string;
+}
```
Comment (kziya) [inline src/datastore/taskService.ts]: Import interfaces from another file, decompose file

File: src/datastore/db.ts [hunk 3]
```diff
@@ -0,0 +1,128 @@
...
+export class DB {
+  index: Index | null = null;
+  path: string;
```
Comment (kziya) [inline src/datastore/db.ts]: Change access modifiers 
```suggestion
  private index: Index | null = null;
  private path: string;
```

File: src/datastore/taskService.ts (hunk 2 above)
Comment (kziya) [inline src/datastore/taskService.ts]:
```suggestion
export type ITaskP = Partial<Omit<ITask, 'done'>>;
```

**Comment by kziya** [comment]
Change file name of `taskService.ts`  to `task.service.ts`

**Review by kziya — APPROVED** [review]

**Review by vladmoroca — APPROVED** [review]

**Pull request merged**